### PR TITLE
adding retry logic on yum commands to avoid resource locking

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:

This PR attempts to address the latest failure in the rpVMSS.sh script by adding retry logic to the yum update commands found therein. Failure can be observed in the ARO production pipeline e2e step.

### What this PR does / why we need it:

Adds retry logic to yum update commands in rpVMSS.sh in order to allow for subsequent attempts if a resource lock is present,.

### Test plan for issue:

<!--
How did you test that this PR works?

E2E will have to prove the logic allows for successful subsequent yum commands. 

### Is there any documentation that needs to be updated for this PR?

No, essentially a knock-on fix of a recent change

### How do you know this will function as expected in production? 

rpVMSS.sh will complete successfully.

### CODE REVIEW

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 1]
```diff
@@ -5,11 +5,18 @@ echo "setting ssh password authentication"
 sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config
 systemctl reload sshd.service
 
-echo "running RHUI fix"
-yum update -y --disablerepo='*' --enablerepo='rhui-microsoft-azure*'
+#Adding retry logic to yum commands in order to avoid stalling out on resource locks
+for attempt in {1..5}; do
+  echo "running RHUI fix"
```
Comment (tsatam) [inline pkg/deploy/generator/scripts/rpVMSS.sh:10]: nit: we should echo this line outside of the loop

File: pkg/deploy/generator/scripts/rpVMSS.sh [hunk 2]
```diff
@@ -5,11 +5,18 @@ echo "setting ssh password authentication"
...
+  yum update -y --disablerepo='*' --enablerepo='rhui-microsoft-azure*'
+  if [[ ${attempt} -lt 5 ]]; then sleep 10; else exit 1; fi
+done
 
-echo "running yum update"
-yum -y -x WALinuxAgent -x WALinuxAgent-udev update --allowerasing
+for attempt in {1..5}; do
+  echo "running yum update"
```
Comment (tsatam) [inline pkg/deploy/generator/scripts/rpVMSS.sh:16]: same change as above

File: pkg/deploy/generator/scripts/rpVMSS.sh (hunk 1 above)
Comment (jhoreman) [reply pkg/deploy/generator/scripts/rpVMSS.sh]: Agreed. Changed and ran make generate

File: pkg/deploy/generator/scripts/rpVMSS.sh (hunk 2 above)
Comment (jhoreman) [reply pkg/deploy/generator/scripts/rpVMSS.sh]: Agreed. Changed and ran make generate

**Review by s-fairchild — APPROVED** [review]
LGTM, legibility changes can be made in a follow up improvement PR.

**Review by hlipsig — APPROVED** [review]

**Pull request merged**